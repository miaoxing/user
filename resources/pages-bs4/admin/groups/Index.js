import React from "react";
import TableProvider from "components/TableProvider";
import Table from "components/bs4/Table";
import Actions from "components/Actions";
import CDeleteLink from "components/CDeleteLink";
import CEditLink from "components/CEditLink";
import CNewBtn from "components/CNewBtn";
import PageHeader from "components/bs4/PageHeader";
import {Button} from "react-bootstrap4";
import app from "app";
import axios from 'axios';

export default class extends React.Component {
  handleClick = (api) => {
    axios.post(app.url('admin/wechat-groups/sync-from-wechat'), {}, {loading: true})
      .then(({data}) => app.ret(data, api.reload));
  };

  render() {
    return <>
      <TableProvider>
        {api => <>
          <PageHeader>
            <div className="float-right">
              {wei.hasWechatGroup && <Button variant="secondary" onClick={this.handleClick.bind(this, api)}>
                从微信同步分组
              </Button>}
              {' '}
              <CNewBtn/>
            </div>
            {wei.page.controllerTitle}
          </PageHeader>

          <Table
            columns={[
              {
                text: '名称',
                dataField: 'name'
              },
              {
                text: '顺序',
                dataField: 'sort',
                sort: true
              },
              {
                text: '状态',
                dataField: 'wechatId',
                hidden: wei.hasWechatGroup,
                formatter: (cell) => cell > 0 ? '已同步' : '未同步',
              },
              {
                text: '操作',
                formatter: (cell, row) => <Actions>
                  <CEditLink id={row.id}/>
                  <CDeleteLink id={row.id}/>
                </Actions>
              },
            ]}
          />
        </>}
      </TableProvider>
    </>;
  }
}
