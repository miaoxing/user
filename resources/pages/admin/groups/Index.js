import React from "react";
import {Table, TableProvider, CTableDeleteLink} from "@miaoxing/table";
import {CEditLink, CNewBtn} from "@miaoxing/clink";
import {Button} from "react-bootstrap";
import {Page, PageActions} from "@miaoxing/page";
import {LinkActions} from "@miaoxing/actions";
import $ from 'miaoxing';
import api from '@miaoxing/api';

export default class extends React.Component {
  state = {};

  constructor(props) {
    super(props);
    this.ref = React.createRef();
  }

  componentDidMount() {
    api.curPath('metadata', {loading: true}).then(ret => this.setState(ret));
  }

  handleClick = (tableApi) => {
    api.post('wechat-groups/sync-form-wechat', {loading: true}).then(ret => $.ret(ret, tableApi.reload));
  };

  render() {
    return <Page>
      <TableProvider>
        {tableApi => <>
          <PageActions>
            {this.state.hasWechatGroup && <Button variant="secondary" onClick={this.handleClick.bind(this, tableApi)}>
              从微信同步分组
            </Button>}
            <CNewBtn/>
          </PageActions>

          <Table
            columns={[
              {
                title: '名称',
                dataIndex: 'name'
              },
              {
                title: '顺序',
                dataIndex: 'sort',
                sorter: true,
              },
              {
                title: '状态',
                dataIndex: 'wechatId',
                hideInTable: !this.state.hasWechatGroup,
                render: text => text > 0 ? '已同步' : '未同步',
              },
              {
                title: '操作',
                dataIndex: 'id',
                render: (id) => (
                  <LinkActions>
                    <CEditLink id={id}/>
                    <CTableDeleteLink id={id}/>
                  </LinkActions>
                )
              },
            ]}
          />
        </>}
      </TableProvider>
    </Page>;
  }
}
