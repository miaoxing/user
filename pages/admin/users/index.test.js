import Index from './index';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import $, {Ret} from 'miaoxing';
import {bootstrap, createPromise, setUrl, resetUrl} from '@mxjs/test';
import {app} from '@mxjs/app';

bootstrap();

describe('admin/users', () => {
  beforeEach(function () {
    setUrl('admin/users');
    app.page = {
      collection: 'admin/users',
      index: true,
    };
  });

  afterEach(() => {
    resetUrl();
    app.page = {};
  });

  test('index', async () => {
    const promise = createPromise();
    const promise2 = createPromise();
    const promise3 = createPromise();

    $.http = jest.fn()
      // 读取地区
      .mockImplementationOnce(() => promise.resolve({
        ret: Ret.suc({
          data: [],
        }),
      }))
      // 读取性别
      .mockImplementationOnce(() => promise2.resolve({
        ret: Ret.suc({
          data: {
            items: [
              {
                id: 0,
                key: 'unknown',
                name: '未知',
              },
              {
                id: 1,
                key: 'male',
                name: '男',
              },
              {
                id: 2,
                key: 'female',
                name: '女',
              },
            ],
          },
        }),
      }))
      // 读取列表数据
      .mockImplementationOnce(() => promise3.resolve({
        ret: Ret.suc({
          data: [
            {
              id: 1,
              name: '姓名1',
              nickName: '昵称2',
              displayName: '显示名称3',
              sex: 1,
              mobile: '138001380000',
              country: '中国',
              province: '广东',
              city: '深圳',
            },
          ],
        }),
      }));

    const {findByText} = render(<MemoryRouter>
      <Index/>
    </MemoryRouter>);

    await findByText('姓名1');
    await findByText('显示名称3');
    await findByText('男');
    await findByText('138001380000');
    await findByText('中国 广东 深圳');

    await Promise.all([promise, promise2, promise3]);
    expect($.http).toHaveBeenCalledTimes(3);
    expect($.http).toMatchSnapshot();
  });
});
