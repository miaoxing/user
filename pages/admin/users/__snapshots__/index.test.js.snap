// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin/users index 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "method": "get",
        "url": "?parentId=0",
      },
    ],
    Array [
      Object {
        "loading": true,
        "method": "get",
        "params": Object {
          "limit": 500,
        },
        "url": "/api/admin/consts/sexConst-sex",
      },
    ],
    Array [
      Object {
        "method": "get",
        "url": "/api/admin/users?page=1&limit=20",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        "reject": [Function],
        "resolve": [Function],
      },
    },
  ],
}
`;
